remix-debug/src/debugger/debugger.js

```diff
@@ -72,31 +72,41 @@ Debugger.prototype.debug = function (blockNumber, txNumber, tx, loadingCb) {
   const self = this
   let web3 = this.debugger.web3
 
-  if (this.debugger.traceManager.isLoading) {
-    return
-  }
+  return new Promise((resolve, reject) => {
+    if (this.debugger.traceManager.isLoading) {
+      return resolve()
+    }
 
-  if (tx) {
-    if (!tx.to) {
-      tx.to = traceHelper.contractCreationToken('0')
+    if (tx) {
+      if (!tx.to) {
+        tx.to = traceHelper.contractCreationToken('0')
+      }
+      self.debugTx(tx, loadingCb)
+      return resolve()
     }
-    return self.debugTx(tx, loadingCb)
-  }
 
-  try {
-    if (txNumber.indexOf('0x') !== -1) {
-      return web3.eth.getTransaction(txNumber, function (_error, result) {
+    try {
+      if (txNumber.indexOf('0x') !== -1) {
+        return web3.eth.getTransaction(txNumber, function (_error, result) {
```

Comment:
If you transform the callback into an array function you won't need to do `self = this`